extensions/interactions/GraphInput/GraphInput.js

```diff
@@ -334,15 +335,19 @@ oppia.directive('graphViz', function() {
           }
           $scope.graph[option] = !$scope.graph[option];
         };
-        $scope.setMode = function(mode, $event) {
-          // Prevents new vertex from being added in add edge mode
-          $event.preventDefault();
-          $event.stopPropagation();
+
+        var setMode = function(mode) {
           $scope.state.currentMode = mode;
           $scope.state.addEdgeVertex = null;
           $scope.state.selectedVertex = null;
           $scope.state.selectedEdge = null;
         };
+        $scope.onClickModeButton = function(mode, $event) {
+          // Prevents new vertex from being added in add edge mode
```

Comment:
Thanks, @tjiang11 for review.
Wouldn't it be better to place it in the definition of a function, at 340 or maybe 341